docs/developer/checkout/performance.mdx

```diff
@@ -28,4 +28,64 @@ You can leverage strategy called "optimistic updates" for that.
 - [React Query](https://tanstack.com/query/v4/docs/framework/react/guides/optimistic-updates)
 - [urql](https://commerce.nearform.com/open-source/urql/docs/graphcache/cache-updates/#optimistic-updates)
 
-In the nutshell, frontend can calculate new price locally and update the UI immediately, while the real response will apply with a delay in the background.
\ No newline at end of file
+In the nutshell, frontend can calculate new price locally and update the UI immediately, while the real response will apply with a delay in the background.
+
+### Lazy vs eager recalculations
+
+Some mutations require price recalculations - such as changing lines quantity or adding a voucher. However, not every mutation behaves the same way.
+
+With performance considerations, there are two strategies how mutations recalculate prices:
+
+- Eager: Mutation will call `*_CALCULATE_TAXES` event during the mutation execution. That means mutation will run slower and the returned value will have latest updates prices values.
+- Lazy: Mutation will mark internal prices to be "dirty", but not call the webhook immediately. Webhook will be called once the value is requested afterward.
```

Comment:
**Question:** it is dirty or invalidated? I'm asking as invalidated may sound more proper than `dirty` 🤔

Reply:
For me dirty is more proper (https://stackoverflow.com/a/1745899) but I believe it may not be common and I can skip introducing this term